hub/core/chunk_engine.py

```diff
@@ -137,15 +143,21 @@ def chunk_id_encoder(self) -> ChunkIdEncoder:
                 )
 
             enc = ChunkIdEncoder()
-            self.cache[key] = enc
+            cache[key] = enc
             return enc
 
-        enc = self.cache.get_cachable(key, ChunkIdEncoder)
+        enc = cache.get_cachable(key, ChunkIdEncoder)
         return enc
 
     @property
     def chunk_id_encoder_exists(self) -> bool:
-        return get_chunk_id_encoder_key(self.key) in self.cache
+        cache = self.mem_cache or self.cache
```

Comment:
why are there 2 cache variables?

Reply:
mem cache keeps the chunk id encoder and tensor meta in memory to prevent it from being written to actual storage by multiple workers. This can be simplified once we get rid of cachable.

Reply:
can we rename to `meta_cache`?

Reply:
@AbhinavTuli 

```python
class ChunkEngine:
    def __init__(...., meta_cache: LRUCache):
        ....
        self._meta_cache = meta_cache
        ....

    @property
    def meta_cache(self)  -> LRUCache:
       return self._meta_cache or self.cache

```